editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1411,7 +1411,13 @@ bool CanvasItemEditor::_gui_input_pivot(const Ref<InputEvent> &p_event) {
 		// Confirm the pivot move
 		if ((b.is_valid() && !b->is_pressed() && b->get_button_index() == BUTTON_LEFT && tool == TOOL_EDIT_PIVOT) ||
 				(k.is_valid() && !k->is_pressed() && k->get_keycode() == KEY_V)) {
-			_commit_canvas_item_state(drag_selection, TTR("Move pivot"));
+			_commit_canvas_item_state(
+					drag_selection,
+					vformat(
+							TTR("Set CanvasItem \"%s\" Pivot Offset to (%d, %d)"),
+							drag_selection[0]->get_name(),
```

Comment:
Are we sure that `drag_selection.size() > 0` in all this code?

Reply:
I don't think the drag_selection can be empty in that part of the code. Unless there's a way to trigger the rotation, pivot dragging, etc... without a selected node. It might happen in some weird buggy situation though.

Reply:
Hm, maybe there should be some `ERR_FAIL_COND(drag_selection.size());` at the start of each helper method, just to be safe?

Note that some of those methods actually mutate `drag_selection` based on filtering `selection` too.

Reply:
Bump. If you think it's safe I don't mind merging as is, but we need some resolution to this comment.